Architecture.md

```diff
@@ -0,0 +1,85 @@
+# Architecture
+
+This is just a proposal of a potential set of classes / services and their responsibilities and what APIs they may have.
+
+## Top-Level Objects
+
+<table>
+  <thead>
+    <tr><th style='width: 120px'>Object</th><th>Description / Responsibility</th><th>Example APIs</th>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Model</td><td>defines the attributes and relationships for resources </td>
+      <td>
+        - save()<br>
```

Comment:
yeah, possibly. I was thinking the models could be a subclass of a `NGModel` defined in this project that would include a reference to the store or something.